UNITTESTS/fakes/ble/GattServerImpl_mock.h

```diff
@@ -57,6 +57,47 @@ class GattServerMock : public ble::impl::GattServer {
     MOCK_METHOD(void, handleDataReadEvent, (const GattReadCallbackParams *params), (override));
     MOCK_METHOD(void, handleEvent, (GattServerEvents::gattEvent_e type, ble::connection_handle_t connHandle, GattAttribute::Handle_t attributeHandle), (override));
     MOCK_METHOD(void, handleDataSentEvent, (unsigned count), (override));
+
+    // Fake part
+    // Descriptor representation of a descriptor registered with ble::test::register_services
+    struct descriptor_t {
+        UUID uuid;
+        ble::attribute_handle_t handle;
+        ble::att_security_requirement_t read_security = ble::att_security_requirement_t::NONE;
+        ble::att_security_requirement_t write_security = ble::att_security_requirement_t::NONE;
+        bool is_readable;
+        bool is_writable;
+        std::vector<uint8_t> value;   // Use capacity to determine the max size.
+    };
+
+    // Characteristic representation of a characteristic registered with ble::test::register_services
+    struct characteristic_t {
```

Comment:
why didn't you use the normal one?

Reply:
The main driver was getting access to the callbacks. but as you pointed out we can add a function for it. 

I started with using normal characteristics as a first prototype but managing the memory was painful because nothing is really owned by characteristics so you need a vector for the characteristics, another for their array, another for their value and it is the same thing for descriptors. It just makes the fake code much harder to read.